exotations/solvers/exotica_control_rrt_solver/src/control_rrt_solver.cpp

```diff
@@ -0,0 +1,190 @@
+//
+// Copyright (c) 2019, University of Edinburgh
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//  * Redistributions of source code must retain the above copyright notice,
+//    this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of  nor the names of its contributors may be used to
+//    endorse or promote products derived from this software without specific
+//    prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+
+#include <exotica_control_rrt_solver/control_rrt_solver.h>
+
+REGISTER_MOTIONSOLVER_TYPE("ControlRRTSolver", exotica::ControlRRTSolver)
+
+namespace exotica
+{
+void ControlRRTSolver::SpecifyProblem(PlanningProblemPtr pointer)
+{
+    if (pointer->type() != "exotica::DynamicTimeIndexedShootingProblem")
+    {
+        ThrowNamed("This ControlRRTSolver can't solve problem of type '" << pointer->type() << "'!");
+    }
+    MotionSolver::SpecifyProblem(pointer);
+    prob_ = std::static_pointer_cast<DynamicTimeIndexedShootingProblem>(pointer);
+    dynamics_solver_ = prob_->GetScene()->GetDynamicsSolver();
+    if (debug_) HIGHLIGHT_NAMED("ControlRRTSolver", "initialized");
+}
+
+void ControlRRTSolver::Setup()
+{
+    int T = prob_->get_T();
+    const int NU = prob_->get_num_controls();
+    const int NX = prob_->get_num_positions() + prob_->get_num_velocities();
+    const double dt = dynamics_solver_->get_dt();
+    if (parameters_.Seed != -1) ompl::RNG::setSeed(parameters_.Seed);
+
+    std::shared_ptr<ob::RealVectorStateSpace> space(
+        std::make_shared<ob::RealVectorStateSpace>(NX));
+
+    state_bounds_ = std::make_shared<ob::RealVectorBounds>(NX);
+
+    for (int i = 0; i < NX; ++i)
```

Comment:
Check that `paramers_.StateLimits` has correct size, otherwise throw

Reply:
Moved that check to `SpecifyProblem`.